.github/workflows/all.yml

```diff
@@ -0,0 +1,284 @@
+name: Deps, Lint, Test, Build, Publish
+on:
+  # Enable manually triggering this workflow via the API or web UI
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+    tags:
+      - v*
+  pull_request:
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  APP_NAME: "k6"
+  DOCKER_IMAGE_ID: ${{ github.repository }}
+
+jobs:
+  deps:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.14.x
+      - name: Check dependencies
+        run: |
+            go version
+            test -z "$(go mod vendor && git status --porcelain)"
+            go mod verify
+
+  lint:
+    runs-on: ubuntu-latest
+    env:
+        GOLANGCI_VERSION: v1.31
+        GO111MODULE: 'on'
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.14.x
+      - name: Install golangci-lint
+        working-directory: /tmp
+        run: go get github.com/golangci/golangci-lint/cmd/golangci-lint@$GOLANGCI_VERSION
+      - name: Run linters
+        run: |
+          BASEREV=$(git merge-base HEAD origin/master)
+          echo "Base revision: $BASEREV"
+          golangci-lint run --out-format=tab --new-from-rev "$BASEREV" ./...
+
+  test:
+    strategy:
+      matrix:
+        go-version: [1.14.x, 1.15.x]
+        platform: [ubuntu-latest, windows-latest]
+    runs-on: ${{ matrix.platform }}
+    env:
+      CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Run tests with code coverage
+        run: |
+          go version
+          export GOMAXPROCS=2
+          export PATH=$GOPATH/bin:$PATH
+          echo "mode: set" > coverage.txt
+          for pkg in $(go list ./... | grep -v vendor); do
+              list=$(go list -test -f  '{{ join .Deps  "\n"}}' $pkg | grep github.com/loadimpact/k6 | grep -v vendor || true)
+              if [ -n "$list" ]; then
+                  list=$(echo "$list" | cut -f1 -d ' ' | sort -u | paste -sd, -)
+              fi
+
+              go test -race -timeout 800s --coverpkg="$list" -coverprofile=$(echo $pkg | tr / -).coverage $pkg
+          done
+          grep -h -v "^mode:" *.coverage >> coverage.txt
+          rm -f *.coverage
+          bash <(curl --fail -s https://codecov.io/bash)
+          go tool cover -html=coverage.txt -o coverage.html
+      - name: Upload coverage report
+        uses: actions/upload-artifact@v2
+        with:
+          name: test-coverage-report
+          path: coverage.html
```

Comment:
This might be a problem... Since this job is run in a matrix of Go versions and platforms, `coverage.html` will be generated and uploaded by 4 different jobs. According to [the upload-artifact documentation](https://github.com/actions/upload-artifact#uploading-to-the-same-artifact), the best case scenario is that we don't know from which environment the coverage was generated in, which given that we skip some tests on Windows might be different than if the last upload was from Linux. The worst case could be corruption, if I guess more than one job uploads at once...

And there's also the issue of uploading the coverage to Codecov 4 times, which should only be done once.

Plus, we probably don't need to run with coverage in all environments, right? This would shorten the time for 3/4 jobs, which is helpful to get quicker feedback, and also use less GHA credits, if they start charging us :)

So I'll split this into a job that runs tests with coverage in a single environment (say, Linux with go1.15) and run tests without coverage on all others. Sounds good?

Reply:
Hhmm for some reason running tests without coverage is always flaky on Windows, as my last 3 attempts failed. 😞 See https://github.com/imiric/k6/runs/1192519114

It was probably relying on the additional delays introduced by the coverage tracing...

@MStoykov Any ideas to quickly address this? Could we maybe run without `-race` or with `-p=1` on Windows if that would lower the chances of this happening?

Reply:
we were previously running without `-race` so I guess we can do that as well now :)

Reply:
codecov has support for .... uh tagging coverage and merging all the tags. But their documentation is currently down ?!? and I am not certain if it was called tags

Reply:
Yeah, that could be an option, as it would be good to have coverage from all platforms. I'll look into it, thanks.

And tests are flaky on Windows even without `-race` :( https://github.com/imiric/k6/runs/1192739335

Reply:
Running with `-p 1` and `GOMAXPROCS=1` seems to work though: https://github.com/imiric/k6/runs/1192809895

Reply:
Here's the build that uploads coverage using [Codecov's flags feature](https://docs.codecov.io/docs/flags): https://github.com/imiric/k6/runs/1193077004

And how it looks in the UI: https://codecov.io/gh/imiric/k6/commit/a384ab1ad3bf283efe86bc1b821f3590fc5bbde0/build

Right now it's running go1.14 jobs without coverage and go.15 jobs with, since I don't think we need coverage for more than one Go version. On Windows I had to run with `GOMAXPROCS=1` and without `-race` in both cases to avoid flakiness.

I'll push the changes to this PR so we can see it here.

Reply:
Looking [at a file it seems awesome](https://codecov.io/gh/imiric/k6/src/a384ab1ad3bf283efe86bc1b821f3590fc5bbde0/ui/summary.go) ... except that it turns out `windows` count empty lines and on `linux` this apparently doesn't happen

Reply:
Yeah, looks like it 🤦  That's a rabbit hole I wouldn't want to get into though... So I'm inclined to leave it as is.

Reply:
OOH ... this is totally not a problem, I couldn't find a golang issue about this as well and decided to leave it :)

Reply:
I'm not sure if this is an issue with Go or with the way we're running the tests on Windows (under MSYS2 right now). A good test would be to run `go test` on PowerShell, but that would involve rewriting the `go list` logic in PS, so... no 😆